library/tests/c_api_test.rs

```diff
@@ -0,0 +1,33 @@
+mod common;
+
+use common::*;
+use tempdir::TempDir;
+use updater::c_api::*;
+
+#[test]
+fn init_with_nulls() {
```

Comment:
nice like the fact that tests can also be separated into their own files 🎉

Reply:
The behavior is different when they are.  They're run file-at-a-time rather than parallel on threads afaict.  They also can only access public functions. They're "integration tests" in rust terms.  I'll probably actually move most of this back once I fix our global config to be per-thread during testing.